android/src/main/java/com/reactnativereceivesharingintent/RNMediaInfoUtils.java

```diff
@@ -0,0 +1,197 @@
+
+                if (displayNameIndex != -1) {
+                    resultMap.put("fileName", queryCursor.getString(displayNameIndex));
+                }
+                if (sizeIndex != -1) {
+                    resultMap.put("fileSize", queryCursor.getString(sizeIndex));
+                }
+                if (dateTakenIndex != -1) {
+                    resultMap.put("DateTimeTaken", queryCursor.getString(dateTakenIndex));
+                }
+                if(dateModifiedIndex != -1){
+                    resultMap.put("DateTimeModified", queryCursor.getString(dateModifiedIndex));
+                }
+                queryCursor.close();
+            }
+            return resultMap;
+        } catch (Exception e) {
+            e.printStackTrace();
+            Log.e(TAG, "Error getting media info", e);
+            return new HashMap<>();
+        }
+
+    }
+
+    private static Map<String, String> getMediaDimensions(Context context, Uri contentUri, Boolean isImage) {
+        try {
+            Map<String, String> resultMap = new HashMap<>();
+            String[] projection = {
+                    MediaStore.MediaColumns.WIDTH,
+                    MediaStore.MediaColumns.HEIGHT
+            };
+            ContentResolver contentResolver = context.getContentResolver();
+            Cursor cursor = contentResolver.query(contentUri, projection, null, null, null);
+
+            if (cursor != null) {
+                cursor.moveToFirst();
+                try {
+                    if (!cursor.isNull(0) && !cursor.isNull(1)) {  // check if dimensions columns are not null
+                        resultMap.put("width", cursor.getString(0));
+                        resultMap.put("height", cursor.getString(1));
+                    }
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+                cursor.close();
+            }
+
+            // If we didn't get dimensions from MediaStore and the media is an image, try ExifInterface
+            if (!resultMap.containsKey("width") && !resultMap.containsKey("height") && isImage) {
+                try {
+                    ExifInterface exif = new ExifInterface(getAbsolutePath(context,contentUri));
+                    String width = exif.getAttribute(ExifInterface.TAG_IMAGE_WIDTH);
+                    String height = exif.getAttribute(ExifInterface.TAG_IMAGE_LENGTH);
+
+                    if (width != null && height != null ) {
+                        resultMap.put("width",width);
+                        resultMap.put("height",height);
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+            return resultMap;
+        } catch (Exception e) {
+            e.printStackTrace();
+            Log.e(TAG, "Error getting media dimensions", e);
+            return new HashMap<>();
```

Comment:
sam to other catch, is it intentional?

Reply:
Yes, our fallback in worst case is empty object